### PR TITLE
Add GoldenCrossEvaluator

### PR DESCRIPTION
**Pull request by valouvaliavlo**

### CODE REVIEW

**Review by Herklos — CHANGES_REQUESTED** [review]
Nice ! Can you integrate it to momentum_evaluator.py instead of creating a new dedicated module please?

**Comment by valouvaliavlo** [comment]
> Nice ! Can you integrate it to momentum_evaluator.py instead of creating a new dedicated module please?

I think trend_evaluator sould be better than momentum

**Comment by Herklos** [comment]
> > Nice ! Can you integrate it to momentum_evaluator.py instead of creating a new dedicated module please?
> 
> I think trend_evaluator should be better than momentum

Ok :+1:

File: Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+DeathAndGoldenCrossEvaluator is based on two [moving averages](https://www.investopedia.com/terms/m/movingaverage.asp), basically one of **50** periods and other one of **200**.
```
Comment (Herklos) [inline Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:1]: :+1:

**Review by Herklos — APPROVED** [review]

File: Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+When the fast moving average cross above the slow moving average it's called a [Golden Cross](https://www.investopedia.com/terms/g/goldencross.asp) and when it's the slow which cross above the fast it's a [Death Cross](https://www.investopedia.com/terms/d/deathcross.asp)
```
Comment (GuillaumeDSM) [inline Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:3]: I think we should add something like:
> Produces a `-1` signal when the fast moving average is above the slow (bull market) one and `1` when the slow moving average is above the fast one (bear market)

File: Evaluator/TA/trend_evaluator/trend.py [hunk 3]
```diff
@@ -26,6 +27,64 @@
...
+
+        ma1 = None
+        ma2 = None
+
+        if self.fast_ma_type == "vwma":
+            ma1 = tulipy.wma(candle_data, volume_data, self.fast_length)[-1]
+        elif self.fast_ma_type == "lsma":
+            ma1 = tulipy.linreg(candle_data, self.fast_length)[-1]
+        else:
+            ma1 = getattr(tulipy, self.fast_ma_type)(candle_data, self.fast_length)[-1]
```
Comment (GuillaumeDSM) [inline Evaluator/TA/trend_evaluator/trend.py:72]: all cases listed in config are valid here, right ?

File: Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md (hunk 2 above)
Comment (valouvaliavlo) [reply Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:3]:
```suggestion
Suggested change 
When the fast moving average cross above the slow moving average(**-1**) this indicate a bull market and it's called a [Golden Cross](https://www.investopedia.com/terms/g/goldencross.asp). When it's the slow which cross above the fast(**1**) it's a [Death Cross](https://www.investopedia.com/terms/d/deathcross.asp) which mean it's a bear market.

```

File: Evaluator/TA/trend_evaluator/trend.py (hunk 3 above)
Comment (valouvaliavlo) [reply Evaluator/TA/trend_evaluator/trend.py:72]: yes !

File: Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md (hunk 2 above)
Comment (GuillaumeDSM) [reply Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:3]: 👍 I think it's also good to say that -1 and 1 are permanent signals, not just when a cross is hapening

Comment (valouvaliavlo) [reply Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:3]: like this ?

```suggestion
If the fast moving average is above the slow moving average this indicate a bull market(**-1**) and when this happened it's called a [Golden Cross](https://www.investopedia.com/terms/g/goldencross.asp).
Inversely, if it's the fast moving average is above the slow moving average this indicate a bear market(**1**) and when this happened it's called a [Death Cross](https://www.investopedia.com/terms/d/deathcross.asp).
```

Comment (GuillaumeDSM) [reply Evaluator/TA/trend_evaluator/resources/DeathAndGoldenCrossEvaluator.md:3]: With minor improvements:

If the fast moving average is above the slow moving average, this indicates a bull market (signal: **-1**) When this happens it's called a [Golden Cross](https://www.investopedia.com/terms/g/goldencross.asp).
Inversely, if it's the fast moving average is above the slow moving average this indicates a bear market (signal: **1**). When this happens it's called a [Death Cross](https://www.investopedia.com/terms/d/deathcross.asp).

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**